### PR TITLE
working @Shareable annotation

### PR DESCRIPTION
**Pull request by shadanan**

### CODE REVIEW

File: elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java [hunk 1]
```diff
@@ -519,6 +526,34 @@ public void addRelation(String fieldName, PersistentResource newRelation) {
...
+            }
+        }
+    }
+
+    /**
+     * Checks if this persistent resource's underlying entity has the @Shareable annotation.
+     *
+     * @return true if this persistent resource's entity has the @Shareable annotation
+     */
+    private boolean hasShareableAnnotation() {
```
Comment (jkusa) [inline elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java]: Should this be named `isShareable` to abstract the use of the annotation?

File: elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java [hunk 2]
```diff
@@ -312,6 +317,7 @@ public boolean updateAttribute(String fieldName, Object newVal) {
     public boolean updateRelation(String fieldName, Set<PersistentResource> resourceIdentifiers) {
         checkPermission(UpdatePermission.class, this);
         checkFieldPermission(UpdatePermission.class, this, fieldName);
+        checkShareable(resourceIdentifiers);
```
Comment (jkusa) [inline elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java]: Can there be a unit test for this case?

File: elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java [hunk 3]
```diff
@@ -505,6 +511,7 @@ public void removeRelation(String fieldName, PersistentResource removeResource)
     public void addRelation(String fieldName, PersistentResource newRelation) {
         checkPermission(UpdatePermission.class, this);
         checkFieldPermission(UpdatePermission.class, this, fieldName);
+        checkShareable(Collections.singleton(newRelation));
```
Comment (jkusa) [inline elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java]: Can there be a unit test for this case?

File: elide-integration-tests/test/java/com/yahoo/elide/endpoints/BookAuthorIT.java [hunk 4]
```diff
@@ -152,6 +169,8 @@ public void testSparseNoFilters() throws Exception {
                         .param("include", "authors")
                         .get("/book").asString());
 
+        System.out.println(responseBody);
```
Comment (jkusa) [inline elide-integration-tests/test/java/com/yahoo/elide/endpoints/BookAuthorIT.java]: Did you want this line?

File: elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java (hunk 2 above)
Comment (shadanan) [inline elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java]: This PR isn't complete. I wanted to allow @aklish to look at the logic. Unit tests forthcoming.

File: elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java (hunk 1 above)
Comment (shadanan) [inline elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java]: Good idea.

File: elide-integration-tests/test/java/com/yahoo/elide/endpoints/BookAuthorIT.java (hunk 4 above)
Comment (shadanan) [inline elide-integration-tests/test/java/com/yahoo/elide/endpoints/BookAuthorIT.java]: Will remove.

File: elide-core/src/main/java/com/yahoo/elide/Elide.java [hunk 5]
```diff
@@ -236,6 +236,7 @@ public ElideResponse patch(
         } catch (HttpStatusException e) {
             return buildErrorResponse(e);
         } catch (ParseCancellationException e) {
+            e.printStackTrace();
```
Comment (aklish) [inline elide-core/src/main/java/com/yahoo/elide/Elide.java]: log - don't print

**Comment by aklish** [comment]
Looks good Shad.  Just the minor comment

**Comment by yahoocla** [comment]
CLA is valid!

**Pull request merged**